src/com/com_nrf24l01_reg.hpp

```diff
@@ -0,0 +1,283 @@
+#ifndef SRC_COM_COM_NRF24L01_REG_HPP_
+#define SRC_COM_COM_NRF24L01_REG_HPP_
+
+#include <array>
+#include <cstdint>
+
+#ifndef UNIT_TEST
+#include "utilities/byte.hpp"
+#else
+#include "byte_mock.hpp"
+#endif
+
+namespace com {
+
+enum class State : std::uint8_t {
+  disabled = 0,
+  enabled
+};
+
+enum class CRCEncodingScheme : std::uint8_t {
+  crc_8bit,
+  crc_16bit
+};
+
+enum class DataPipe : std::uint8_t {
+  rx_pipe_0 = 0,
+  rx_pipe_1,
+  rx_pipe_2,
+  rx_pipe_3,
+  rx_pipe_4,
+  rx_pipe_5,
+  tx_pipe
+};
+
+enum class DataPipeAddressWidth : std::uint8_t {
+  aw_3bytes = 0x1,
+  aw_4bytes = 0x2,
+  aw_5bytes = 0x3,
+};
+
+enum class AutoRetransmissionDelay : std::uint8_t {
+  // auto retransmis delay 250-4000 us
+  ard250us = 0x0,
+  ard500us = 0x1,
+  ard750us = 0x2,
+  ard1000us = 0x3,
+  ard1250us = 0x4,
+  ard1500us = 0x5,
+  ard1750us = 0x6,
+  ard2000us = 0x7,
+  ard2250us = 0x8,
+  ard2500us = 0x9,
+  ard2750us = 0xa,
+  ard3000us = 0xb,
+  ard3250us = 0xc,
+  ard3500us = 0xd,
+  ard3750us = 0xe,
+  ard4000us = 0xf
+};
+
+enum class AutoRetransmitCount : std::uint8_t {
+  // auto retransmit count 0-15 in case of failed AR
+  arc0 = 0x0,
+  arc1 = 0x1,
+  arc2 = 0x2,
+  arc3 = 0x3,
+  arc4 = 0x4,
+  arc5 = 0x5,
+  arc6 = 0x6,
+  arc7 = 0x7,
+  arc8 = 0x8,
+  arc9 = 0x9,
+  arc10 = 0xa,
+  arc11 = 0xb,
+  arc12 = 0xc,
+  arc13 = 0xd,
+  arc14 = 0xe,
+  arc15 = 0xf
+};
+
+enum class OperationMode : std::uint8_t {
+  prim_rx = 0,
+  prim_tx
+};
+
+enum class RFPowerSetting : std::uint8_t {
+  rf_pwr_18dBm = 0x0,
+  rf_pwr_12dBm = 0x1,
+  rf_pwr_6dBm = 0x2,
+  rf_pwr_0dBm = 0x3
+};
+
+enum class DataRateSetting : std::uint8_t {
+  rf_dr_1mbps = 0,
+  rf_dr_2mbps = 1
+};
+
+using data_pipe_address = std::array<std::uint8_t, 5>;
+using register_t = std::uint8_t;
+
+namespace instruction_word {
+static constexpr std::uint8_t R_REGISTER = 0x00;
+static constexpr std::uint8_t W_REGISTER = 0x20;
+static constexpr std::uint8_t R_RX_PAYLOAD = 0x61;
+static constexpr std::uint8_t W_TX_PAYLOAD = 0xa0;
+static constexpr std::uint8_t FLUSH_TX = 0xe1;
+static constexpr std::uint8_t FLUSH_RX = 0xe2;
+static constexpr std::uint8_t REUSE_TX_PL = 0xe3;
+static constexpr std::uint8_t NOP = 0xff;
+}  // namespace instruction_word
+
+namespace rf_config {
+static constexpr std::uint8_t rf_channel = 20;
+}
+
+namespace reg {
+/// configuration register
+namespace config {
+static constexpr std::uint8_t REG_ADDR = 0x0;
+static constexpr std::uint8_t MASK_RX_DR = 6;
+static constexpr std::uint8_t MASK_TX_DS = 5;
+static constexpr std::uint8_t MASK_MAX_RT = 4;
+static constexpr std::uint8_t EN_CRC = 3;
+static constexpr std::uint8_t CRCO = 2;
+static constexpr std::uint8_t PWR_UP = 1;
+static constexpr std::uint8_t PRIM_RX = 0;
+}  // namespace config
+
+/// enable enhanced shock burst auto acknowledgement
+namespace en_aa {
```

Comment:
dangerously short please make the name more readable.

Reply:
Naming taken from nrf24l01 documentation. Will leave it as agreed.